packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -538,6 +539,86 @@ export class ConnectionService {
     }
   }
 
+  /**
+   * If knownConnectionId is passed, it will compare the incoming connection id with the knownConnectionId, and skip the other validation.
+   *
+   * If no known connection id is passed, it asserts that the incoming message is in response to an attached request message to an out of band invitation.
+   * If is the case, and the state of the out of band record is still await response, the state will be updated to done
+   *
+   */
+  public async matchIncomingMessageToRequestMessageInOutOfBandExchange(
```

Comment:
Wow this is a huge method with a huge name!

Reply:
Yeah, i like explicitness over being brief. But suggestions welcome! :)